mlir/include/mlir/Dialect/Rock/Tuning/RockTuningTypes.h

```diff
@@ -0,0 +1,41 @@
+//===--------- ConvContext.h - MLIR tuning parameter generation ----------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines MLIR base types for tuning
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_ROCK_ROCKTUNINGTYPE_H
+#define MLIR_DIALECT_ROCK_ROCKTUNINGTYPE_H
+
+#include "mlir/Dialect/Rock/IR/Rock.h"
+
+namespace mlir {
+namespace rock {
+
+enum TunableOpType { TunableOpConv2dFwd, TunableOpGemm };
```

Comment:
Nit: `enum class`

Also, let's not make a separate enum here. Could you please instead extend `ConvOpType` to `GemmWrapperOpType` and add `Gemm` as a case, or make a second enum `GemmWrapperOpType` that includes all the convolutions? Then, you can change the API in the tuning parameters struct to take this four-way enum - I'm wanting this for `miopen-gen` anyhow